sklearn/neighbors/tests/test_ball_tree.py

```diff
@@ -85,3 +85,13 @@ def test_array_object_type():
     X = np.array([(1, 2, 3), (2, 5), (5, 5, 1, 2)], dtype=object)
     with pytest.raises(ValueError, match="setting an array element with a sequence"):
         BallTree(X)
+
+
+def test_bad_pyfunc_metric():
+    def wrong_distance(x, y):
+        return "1"
```

Comment:
I would check for both: One function that does not return a float and another function with only one argument (as before).